discount/src/main/java/com/kyriba/conference/discount/api/DiscountController.java

```diff
@@ -1,66 +1,58 @@
 package com.kyriba.conference.discount.api;
 
-import com.kyriba.conference.discount.api.dto.DiscountDto;
+import com.kyriba.conference.discount.api.dto.DiscountExternalDto;
+import com.kyriba.conference.discount.api.dto.DiscountResponse;
 import com.kyriba.conference.discount.api.dto.DiscountUpdateParamsDto;
-import com.kyriba.conference.discount.domain.DiscountType;
-import io.swagger.annotations.ApiModel;
-import io.swagger.annotations.ApiModelProperty;
+import com.kyriba.conference.discount.service.DiscountService;
 import io.swagger.annotations.ApiParam;
-import lombok.Value;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.*;
 
 import javax.validation.Valid;
-import javax.validation.constraints.NotNull;
-import java.util.Arrays;
 import java.util.List;
 
-import static com.kyriba.conference.discount.domain.DiscountType.JUNIOR;
-import static com.kyriba.conference.discount.domain.DiscountType.STUDENT;
-
 /**
  * @author Igor Lizura
  */
 @RestController
 @RequestMapping(value = "/api/v1/discounts")
 public class DiscountController {
+    private final DiscountService discountService;
+
+    public DiscountController(DiscountService discountService) {
```

Comment:
Could be simplified with AllArgsConstructor annotation.

Reply:
fixed